querybook/webapp/components/AIAssistant/AutoFixButton.tsx

```diff
@@ -21,26 +20,31 @@ interface IProps {
 
 const useSQLFix = () => {
     const [data, setData] = useState<{ [key: string]: string }>({});
+    const [prevFixedQuery, setPrevFixedQuery] = useState('');
 
     const socket = useAISocket(AICommandType.SQL_FIX, ({ data }) => {
-        setData(data);
+        setData(data as { [key: string]: string });
     });
 
     const {
         data: unformattedData,
         explanation,
         fix_suggestion: suggestion,
-        fixed_query: rawFixedQuery,
+        fixed_query: fixedQuery,
     } = data;
 
-    const fixedQuery = trimSQLQuery(rawFixedQuery);
+    useEffect(() => {
```

Comment:
could we generalize this a bit more to handle the json streaming unstability, like
```ts
function useNonEmptyState<T>(initValue: T | () => T) => {
    const [state, _setState] = useState(initValue);
    const setState = useCallback((newValOrFunc: T | (oldVal: T) => T) => {
         _setState(oldVal => {
             let newVal: T = newValOrFunc;
             if (typeof newValOrFunc === 'function')
                 newVal = newValOrFunc(oldVal);
             return newVal || oldVal;
         });     
    }, []);
    return [state, setState];
}
```